packages/fluentui/react-builder/src/components/List.tsx

```diff
@@ -14,7 +14,7 @@ export type ListProps = {
 export const List: React.FunctionComponent<ListProps> = ({ onDragStart, style }) => {
   const [filter, setFilter] = React.useState<string>('');
 
-  const filterRegexp = new RegExp(filter, 'i');
+  const filterRegexp = React.useMemo(() => new RegExp(filter, 'i'), [filter]);
```

Comment:
The improved hooks rule pointed out that this regex being created on every render was causing some later hook to re-run on every render